Removing gnacs dependency

fionapigott opened the pull request:
I've removed all gnacs dependencies.

- Update setup.py to reflect the new dependency (tweet_parser)
- Edit the code to use tweet_parser properties anywhere it accesses a value of a tweet
- compatibility with Python 2 and Python 3
- tested by adding my API creds to test_search.sh and running bash test_search.sh
- updated the version number to 1.2.0 (last version on pypi is 1.1.0)

fionapigott commented:
Also, once this is merged we should get added to the "gapi" pypi package so that we can update that.

jeffakolb commented on gnip_search.py:
```diff
@@ -74,8 +78,8 @@ def __init__(self, token_list_size=40):
             sys.exit(-1)
 
         # Gnacs is not yet upgraded to python3, so don't allow CSV output option (which uses Gnacs) if python3
```
Do we want references to Gnacs in here?

fionapigott replied:
good call, I changed that comment

fionapigott commented:
I removed references to search V1 vs V2, and changed the URL validation to allow data-api urls as well as gnip-api urls (however, the "gnip-api" urls are the official url, so I only suggest that domain in the error message)
